src/Controller/DashboardController.php

```diff
@@ -195,6 +200,54 @@ public function rssAction($roomId, Request $request)
         return array(
         );
     }
+
+    /**
+     * @Route("/dashboard/{roomId}/myviews")
+     * @Template()
+     */
+    public function myViewsAction($roomId, Request $request, LegacyEnvironment $legacyEnvironment)
+    {
+        $searchData = new SearchData();
+
+        // get the current user's saved searches
+        $em = $this->getDoctrine()->getManager();
+        $repository = $em->getRepository(SavedSearch::class);
+        $currentUser = $legacyEnvironment->getEnvironment()->getCurrentUserItem();
+        $portalUserId = $currentUser->getRelatedPortalUserItem()->getItemId();
+
+        $savedSearches = $repository->findByAccountId($portalUserId);
+        $searchData->setSavedSearches($savedSearches);
+
+        $myViewsForm = $this->createForm(MyViewsType::class, $searchData, [
+            'action' => $this->generateUrl('app_dashboard_myviews', [
+                'roomId' => $roomId
+            ])
+        ]);
+        $myViewsForm->handleRequest($request);
+
+        if ($myViewsForm->isSubmitted() && $myViewsForm->isValid()) {
+
+            $savedSearch = $searchData->getSelectedSavedSearch();
+
+            if ($savedSearch) {
+                $savedSearchURL = $savedSearch->getSearchUrl();
+                if ($savedSearchURL) {
+                    // redirect to the search_url stored for the chosen saved search
+                    $redirectResponse = new RedirectResponse($request->getSchemeAndHttpHost() . $savedSearchURL);
+
+                    return $redirectResponse;
```

Comment:
Hier wäre ein redirectToRoute besser, aber dann bräuchte es eine Array-Repräsentation der URL.

```suggestion
                    return $this->redirect($request->getSchemeAndHttpHost() . $savedSearchURL);
```

Reply:
wenn wir `redirectToRoute()` verwenden, bräuchte es nicht nur eine Array-Repräsentation aller Parameter des Query Strings, sondern müsste auch die ID des Raumes mit erfasst werden (entweder als Teil der Array-Repräsentation oder als separate `context_id` Spalte in der `saved_searches` SQL-Tabelle). Und wir müssten das Parsen+Dekodieren in die Array-Repräsentation sowie das Generieren+Enkodieren in den URL-String selbst übernehmen. Diese unnötigen und potentiell fehleranfälligen Mehraufwand würde ich gerne vermeiden.

Zudem ist die Request URI quasi eine kompakte String-Repräsentation aller Parameter, die so wie sie ist als Bestandteil der URL verwendet werden kann.. Der Nachteil ist, dass die Route (`…/search/results/…`) in den gespeicherten Suchen mit gespeichert wird und es bei einer Änderung der Route eine Migration bräuchte. Dennoch würde ich gerne vorerst bei dem aktuellen Setup bleiben.

Ich habe bewusst `return $redirectResponse;` und nicht `return new RedirectResponse(...)` verwendet, da man so nachher beim Debuggen einen Breakpoint auf `$redirectResponse` setzen kann, ohne den Code umschreiben zu müssen. Dies würde ich also auch gerne so belassen.